kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/validation/impl/AssetGlobalRule.java

```diff
@@ -0,0 +1,88 @@
+package edu.arizona.kfs.module.cam.document.validation.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.module.cam.businessobject.AssetGlobal;
+import org.kuali.kfs.module.cam.businessobject.AssetGlobalDetail;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.arizona.kfs.module.cam.CamsConstants;
+import edu.arizona.kfs.module.cam.CamsKeyConstants;
+import edu.arizona.kfs.module.cam.CamsPropertyConstants;
+import edu.arizona.kfs.module.cam.businessobject.AssetGlobalDetailExtension;
+import edu.arizona.kfs.module.cam.businessobject.AssetInventoryUnit;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+
+public class AssetGlobalRule extends org.kuali.kfs.module.cam.document.validation.impl.AssetGlobalRule {
+
+	@Override
+	protected boolean processCustomSaveDocumentBusinessRules(MaintenanceDocument document) {
+		boolean success = true;
+		success &= super.processCustomSaveDocumentBusinessRules(document);
```

Comment:
Before proceeding with our validations, you should check to see if the super call returned any errors:
`if (GlobalVariables.getMessageMap().hasErrors()) {
      return false;
}`
See org.kuali.kfs.module.cam.document.validation.impl.AssetGlobalRule, lines 866-868.